coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java

```diff
@@ -121,35 +129,39 @@ public SimpleClusterManager(CoordinatorConf conf, Configuration hadoopConf) thro
   void nodesCheck() {
     try {
       long timestamp = System.currentTimeMillis();
-      Set<String> deleteIds = Sets.newHashSet();
-      Set<String> unhealthyNode = Sets.newHashSet();
       for (ServerNode sn : servers.values()) {
         if (timestamp - sn.getTimestamp() > heartbeatTimeout) {
           LOG.warn("Heartbeat timeout detect, " + sn + " will be removed from node list.");
-          deleteIds.add(sn.getId());
+          sn.setStatus(ServerStatus.LOSTED);
+          lostNodes.add(sn);
```

Comment:
Should we remove the node in the unhealthyNodes if an unhealthy node is lost?

Reply:
We should remove the node in the lostNodes too if an lost node become unhealthy. So making unhealthyNodes as a local variable will make it more simple.

Reply:
Good advice